### PR TITLE
AFM cache volume fixes

### PR DESCRIPTION
**Pull request by amdabhad**
## Pull request checklist

AFM cache volume fixes for:
- Delete bucket keys in volume deletion flow. **Note: This requires Scale 5.2.1 on storage cluster** (If you are using Scale 5.2.1 dev build, make sure to use the build from 4th July or later)
- Use SSL option by default if endpoint is https
- Remaining pieces of keys masking

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature Enhancement
- [ ] Test Automation 
- [ ] Code Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Community Operator listing 
- [ ] Other (please describe): 

## How risky is this change?
- [ ] Small, isolated change
- [x] Medium, requires regression testing
- [ ] Large, requires functional and regression testing

### CODE REVIEW

**Comment by amdabhad** [comment]
CSI driver image: quay.io/amdabhad/csi-driver:afmCacheFixes0907

File: driver/csiplugin/utils/http_utils.go [hunk 1]
```diff
@@ -78,7 +84,13 @@ func HttpExecuteUserAuth(ctx context.Context, httpClient *http.Client, requestTy
 	request.Header.Add("Accept", "application/json")
 
 	request.SetBasicAuth(user, password)
-	klog.V(6).Infof("[%s] http_utils HttpExecuteUserAuth request: %+v", GetLoggerId(ctx), request)
+
+	logLevel := strings.ToUpper(os.Getenv(LogLevel))
+	if strings.Contains(requestURL, BucketKeysURL) && logLevel == TRACE.String() && request != nil {
```
Comment (hemalathagajendran) [inline driver/csiplugin/utils/http_utils.go:89]: if we restrict the condition to AFM cache feature then for normal cases we might not see this log.

Comment (amdabhad) [reply driver/csiplugin/utils/http_utils.go]: Right, fixed it

Comment (amdabhad) [reply driver/csiplugin/utils/http_utils.go]: @saurabhwani5, the driver image with this log change is updated [here](https://github.com/IBM/ibm-spectrum-scale-csi/pull/1181#issuecomment-2217235612)

**Review by hemalathagajendran — APPROVED** [review]

**Pull request merged**